test/brsTypes/components/RoSGNode.test.js

```diff
@@ -986,6 +986,741 @@ describe("RoSGNode", () => {
+
+                appendChild.call(interpreter, child1);
+                appendChild.call(interpreter, child2);
+
+                let childrenInserted = insertChildren.call(
+                    interpreter,
+                    new RoArray([child1, child4]),
+                    new Int32(-11)
+                );
+                let result = getChildren.call(interpreter, new Int32(-1), new Int32(0));
+                expect(childrenInserted.value).toEqual(true);
+                expect(result.elements.length).toEqual(3);
+                expect(result.elements[0]).toEqual(child2);
+            });
+            it("inserts new and existing nodes with negative index with rollover", () => {
+                let appendChild = parent.getMethod("appendchild");
+                let getChildren = parent.getMethod("getchildren");
+                let insertChildren = parent.getMethod("insertchildren");
+
+                appendChild.call(interpreter, child1);
+                appendChild.call(interpreter, child2);
+                appendChild.call(interpreter, child3);
+
+                let childrenInserted = insertChildren.call(
+                    interpreter,
+                    new RoArray([child2, child4]),
+                    new Int32(-1)
+                );
+                let result = getChildren.call(interpreter, new Int32(-1), new Int32(0));
+                expect(childrenInserted.value).toEqual(true);
+                expect(result.elements.length).toEqual(4);
+                expect(result.elements[0]).toEqual(child4);
+                expect(result.elements[3]).toEqual(child2);
+            });
+            it("returns false for empty array", () => {
+                let appendChild = parent.getMethod("appendchild");
+                let getChildren = parent.getMethod("getchildren");
+                let insertChildren = parent.getMethod("insertchildren");
+
+                appendChild.call(interpreter, child1);
+                appendChild.call(interpreter, child2);
+
+                let childrenInserted = insertChildren.call(
+                    interpreter,
+                    new RoArray([]),
+                    new Int32(-1)
+                );
+                let result = getChildren.call(interpreter, new Int32(-1), new Int32(0));
+                expect(childrenInserted.value).toEqual(false);
+                expect(result.elements.length).toEqual(2);
+            });
+        });
+        describe("getchild", () => {
+            it("gets an existing child", () => {
+                let appendChild = parent.getMethod("appendchild");
+                let getChild = parent.getMethod("getchild");
+
+                appendChild.call(interpreter, child1);
+                appendChild.call(interpreter, child2);
+
+                let existingChild = getChild.call(interpreter, new Int32(1));
+                expect(existingChild).toBeInstanceOf(RoSGNode);
```

Comment:
Should we add an `expect` here that this returns the child we want it to?

Reply:
Oh yeah good call